### ISSUE TITLE
.venv directory spuriously copied with exports_sources

### ISSUE THREAD

**Issue by mattgodbolt** [conan-io/conan#3535]
I'm running on Ubuntu Linux 18.04 with conan 1.7.0, run from a virtualenv in ~/.conan-venv/bin/conan which only contains conan.

My project is header only, and as part of its build makes a new fresh `.venv`. This `.venv` directory is _not_ listed in the `exports_sources` but is getting erroneously published (and causes trouble on other people's machines).

Repro steps:
```
/t/conan-bug $ ls -la
total 28
drwxr-xr-x   3 mattgodbolt mattgodbolt  4096 Sep 12 10:49 ./
drwxrwxrwt 153 root        root        16384 Sep 12 10:49 ../
-rw-r--r--   1 mattgodbolt mattgodbolt   266 Sep 12 10:45 conanfile.py
drwxr-xr-x   2 mattgodbolt mattgodbolt  4096 Sep 12 10:39 header/

/t/conan-bug $ ls header
header.h

/t/conan-bug $ cat conanfile.py 
from conans import ConanFile

class Test(ConanFile):
    name = "test"
    version = "0.0"
    url = ""
    description = "test"
    exports_sources = "header/*"
    no_copy_source = True

    def package(self):
        self.copy("*.h", dst="include", src="header")

/t/conan-bug $ virtualenv .venv
Running virtualenv with interpreter /usr/bin/python2
New python executable in /tmp/conan-bug/.venv/bin/python2
Also creating executable in /tmp/conan-bug/.venv/bin/python
...etc...

$ conan create . test/0.0@mattgodbolt/testing
...
/t/conan-bug $ ls ~/.conan/data/test/0.0/mattgodbolt/testing/export_source/ -la
total 16
drwxr-xr-x 4 mattgodbolt mattgodbolt 4096 Sep 12 10:45 ./
drwxr-xr-x 8 mattgodbolt mattgodbolt 4096 Sep 12 10:45 ../
drwxr-xr-x 2 mattgodbolt mattgodbolt 4096 Sep 12 10:45 header/
drwxr-xr-x 4 mattgodbolt mattgodbolt 4096 Sep 12 10:45 .venv/
```

the `.venv` is copied. Disabling `exports_sources` in the conanfile "fixes" this. Also it seems the `.venv` has to actually be a virtualenv: if it's just a directory with "empty.txt" in it, it doesn't get copied.

**Comment by mattgodbolt** [comment]
Renaming the `.venv` directory doesn't fix anything: somehow `conan` is picking up something about the virtual env directory. I tried to repro with things like subdirectories with symlinks outside of the build folder...no joy. Unusual one :)

**Comment by mattgodbolt** [comment]
Ok got it!

Seemingly a symlink to a directory outside of the build will be always copied!

Same conanfile as above, fresh clean directory:

```
/t/conan-bug2 $ ls -l
total 8
-rw-r--r-- 1 mattgodbolt mattgodbolt  266 Sep 12 11:08 conanfile.py
drwxr-xr-x 2 mattgodbolt mattgodbolt 4096 Sep 12 11:08 header/
/t/conan-bug2 $ cat conanfile.py 
from conans import ConanFile

class Test(ConanFile):
    name = "test"
    version = "0.0"
    url = ""
    description = "test"
    exports_sources = "header/*"
    no_copy_source = True

    def package(self):
        self.copy("*.h", dst="include", src="header")
```

then:

```
/t/conan-bug2 $ ln -sf /usr 
/t/conan-bug2 $ conan create . test/0.0@mattgodbolt/testing
...
```
the `usr` symlink is copied!

```
~ $ ls -l ~/.conan/data/test/0.0/mattgodbolt/testing/export_source/
    
total 4
drwxr-xr-x 2 mattgodbolt mattgodbolt 4096 Sep 12 11:09 header/
lrwxrwxrwx 1 mattgodbolt mattgodbolt    4 Sep 12 11:09 usr -> /usr/
```

**Comment by memsharded** [comment]
It has been fixed in #3537, it will be released in conan 1.8, but if you could try it running from source, "develop" branch, that would also help. Thanks!
